agent/terraform/README.md

```diff
@@ -0,0 +1,70 @@
+# Deploying with Terraform + Juju
+
+This Terraform module can be used for deploying the Testflinger agent
+host on a physical or virtual machine model using Juju.
+
+## Deploying an Environment
+
+This terraform module assume that the Juju model you want to deploy
+to has already been created.
+
+1. First, create the model
+```
+    $ juju add-model agent-host-1
+```
+
+2. Create ssh keys to use on the agent host
+```
+    $ ssh-keygen -t rsa -f mykey
+```
+
+3. Create a git repo with the Testflinger configs
+
+If you have more than one agent host to deploy, create a separate directory
+for each of them in this repo. Then create a separate directory for each
+agent in the agent host directory where they reside. For example:
+
+```
+/agent-host-1
+-/agent-101
+  -testflinger-agent.yaml
+  -default.conf
+-/agent-102
+  -testflinger-agent.yaml
+  -default.conf
+/agent-host-2
+...
+```
+
+4. Create a main.tf which specifies the required parameters for this module
+
+```
+terraform {
+  required_providers {
+    juju = {
+      version = "~> 0.13.0"
+      source  = "juju/juju"
+    }
+  }
+}
```

Comment:
perhaps there's no need to add this to `main.tf` given that it's already in `versions.tf`

Reply:
I thought so too, but if I leave it out then I get an error:
```
╷
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider hashicorp/juju: provider registry registry.terraform.io does not have a provider
│ named registry.terraform.io/hashicorp/juju
│ 
│ Did you intend to use juju/juju? If so, you must specify that source address in each module which requires that provider. To see which modules
│ are currently depending on hashicorp/juju, run the following command:
│     terraform providers
```
Not sure if this is something new in the latest versions of the terraform provider or what, because I didn't think I previously had to do this.

Reply:
That's strange, I also recall this working before. I guess it doesn't hurt to have it if it doesn't work otherwise